spmd/tensor/device_mesh.py

```diff
@@ -249,40 +244,61 @@ def get_coordinate_on_dim(self, dim: int) -> Optional[int]:
         return self._coordinate_on_dim[dim] if self._coordinate_on_dim else None
 
     def scatter(
-        self, scatter_list: List[torch.Tensor], mesh_dim: int = 0
+        self,
+        tensor_to_scatter: torch.Tensor,
+        mesh_dim: int = 0,
+        tensor_dim: int = 0,
     ) -> torch.Tensor:
         """
-        scatter the list of tensors to a device mesh dimension. We by default
+        scatter a big tensor to a device mesh dimension. We by default
         use the first rank of the mesh dimension as the source of truth, i.e
         for a 2d mesh [[0, 1], [2, 3]], if we scatter on mesh_dim = 1, we will
-        scatter the tensor list on rank 0 to rank 0/1, and tensor list on rank 2
-        to rank 2/3.
+        scatter the tensor splitted on rank 0 to rank 0/1, and tensor splitted
+        on rank 2 to rank 2/3.
 
         Args:
-            scatter_list (List[torch.Tensor]): list of tensors to scatter.
+            tensor_to_scatter (torch.Tensor): the tensor to be scattered.
             mesh_dim (int, optional): indicate which mesh dimension we want
                 to scatter on, we by default choose the first rank on the
-                mesh dimension.
+                mesh dimension as source of truth.
+            tensor_dim (int, optional): indicate which tensor dimension we want
+                to split the `tensor_to_scatter` before scattering.
 
         Returns:
             A :class:`torch.Tensor` object
         """
+        my_coordinate = self.get_coordinate_on_dim(mesh_dim)
+        # TODO: what should happen if rank is not in the mesh?
+        assert (
+            my_coordinate is not None
+        ), "Rank if not part of mesh"  # TODO: figure out behavior here
+
+        num_chunks = self.size(mesh_dim)
+        # TODO: handle uneven shard sizes
+        assert tensor_to_scatter.size(tensor_dim) % num_chunks == 0, (
+            f"Only support chunk sharding evenly now, but tensor got "
+            f"dimension {tensor_dim} of size {tensor_to_scatter.size(tensor_dim)}, "
+            f"which does not divide number of shards {num_chunks}."
+        )
+
+        scatter_list = list(
+            tensor_to_scatter.tensor_split(num_chunks, dim=tensor_dim)
```

Comment:
Interesting, I didn't know that PyTorch also support this. We might want to document this somewhere. (I mean this is different from the default behavior of Chunk. What if users want the same behavior of Chunk? So we can set the expectation right here.)

Reply:
yeah added to the doc of `distribute_tensor`